docs/build/cookbook.rst

```diff
@@ -791,7 +791,7 @@ any operations::
         # ...
 
         def process_revision_directives(context, revision, directives):
-            if config.cmd_opts.autogenerate:
```

Comment:
config is present in env.py:

https://github.com/sqlalchemy/alembic/blob/master/alembic/templates/generic/env.py#L11

additionally this could at least be context.config

why are you getting an error?

Reply:
I am getting error while trying to create revision programaticaly, not via CLI. `cmd_opts` are `None` then. I will try tommorow (CET) `context.config` with my use case.

Reply:
so the recipe here is given as one someone uses in their env.py, I'm not sure how you'd use it outside of env.py but we can clarify that